File: 125-valid-palindrome/valid-palindrome.cpp
class Solution {
public:
    bool isPal(string &str, int i) {
        if (i >= str.length() / 2) return true;
        if (tolower(str[i]) != tolower(str[str.length() - 1 - i])) return false;
        return isPal(str, i + 1);
    }

    bool isPalindrome(string s) {
        string str = "";
        for (int i = 0; i < s.length(); i++) {
            if (isalnum(s[i])) str += s[i];
        }
        return isPal(str, 0);

    }
};

File: 125-valid-palindrome/valid-palindrome.java
class Solution {
    public boolean isPalindrome(String s) {
        StringBuilder temp = new StringBuilder();
        for (int i = 0; i < s.length(); i++) {
            if (Character.isLetterOrDigit(s.charAt(i))) {
                temp.append(Character.toLowerCase(s.charAt(i)));
            }
        }
        int i = 0;
        int j = temp.length() - 1;
        while (i != temp.length() / 2) {
            if (temp.charAt(i++) != temp.charAt(j--)) {
                return false;
            }
        }
        return true;
    }
}